pkg/assets/builder.go

```diff
@@ -127,32 +142,123 @@ func (a *AssetBuilder) RemapImage(image string) (string, error) {
 	}
 
 	a.ContainerAssets = append(a.ContainerAssets, asset)
-
 	return image, nil
 }
 
 // RemapFile sets a new url location for the file, if a AssetsLocation is defined.
-func (a AssetBuilder) RemapFile(file string) (string, error) {
-	if file == "" {
-		return "", fmt.Errorf("unable to remap an empty string")
+func (a *AssetBuilder) RemapFileAndSHA(fileURL *url.URL) (*url.URL, *hashing.Hash, error) {
+	if fileURL == nil {
+		return nil, nil, fmt.Errorf("unable to remap an empty URL")
+	}
+
+	fileAsset := &FileAsset{
+		FileURL: fileURL,
+	}
+
+	if a.AssetsLocation != nil && a.AssetsLocation.FileRepository != nil {
+		fileAsset.CanonicalFileURL = fileURL
+
+		normalizedFileURL, err := a.normalizeURL(fileURL)
+		if err != nil {
+			return nil, nil, fmt.Errorf("unable to parse file url %q: %v", normalizedFileURL, err)
```

Comment:
nit: "...to normalize file url..."

But also you need to log `fileURL`, as normalizedFileURL is presumably nil

Reply:
sorta ... fixed it nice catch

Reply:
Actually fmt.Errorf is in the wrong place